### PR TITLE
Added class parsing

### PR DESCRIPTION
**Pull request by ryanmcnamara**
Fixes https://github.com/giladgray/ts-quick-docs/issues/5

### CODE REVIEW

File: src/documentation.ts [hunk 1]
```diff
@@ -60,6 +60,10 @@ export default class Documentation {
                 // This is a top level interface, get its symbol
                 const symbol = this.checker.getSymbolAtLocation((node as ts.InterfaceDeclaration).name);
                 output.push(this.serializeInterface(symbol, this.getFileName(node)));
+            } else if (node.kind === ts.SyntaxKind.ClassDeclaration) {
+                const delcaration = node as ts.ClassDeclaration;
```
Comment (giladgray) [inline src/documentation.ts]: # TYPO SHUT IT DOWN

File: src/documentation.ts [hunk 2]
```diff
@@ -60,6 +60,10 @@ export default class Documentation {
...
+                const symbol = this.checker.getSymbolAtLocation((node as ts.ClassDeclaration).name);
```
Comment (giladgray) [inline src/documentation.ts]: you already casted this: `declaration.name`

File: src/documentation.ts [hunk 3]
```diff
@@ -141,6 +145,28 @@ export default class Documentation {
...
+            tags: this.getJsDocTags(symbol),
+            type: "class",
+        };
+
+        const classNode = symbol.declarations[0] as ts.ClassDeclaration;
+        if (classNode.heritageClauses != null) {
+            details.extends = classNode.heritageClauses[0].types.map((type) => type.getText());
+        }
+
+        // Get the props signatures
```
Comment (giladgray) [inline src/documentation.ts]: "member signatures"

big question: how similar are members to props? are there things that members can do that props can't? what about `private/public`?

File: test/fixtures/class.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/** A class */
+export class Class {
+    /** text value */
+    public value: string;
```
Comment (giladgray) [inline test/fixtures/class.ts]: please test `private` fields too

File: src/documentation.ts (hunk 1 above)
Comment (giladgray) [inline src/documentation.ts]: also this variable is unused?

**Comment by giladgray** [comment]
@ryanmcnamara waiting on a push from you to address comments. 

also can you enable circle? or, if you'd prefer, i can add you as a collaborator and you can work directly in this repo instead of forking. then you can leverage my existing circle setup.

**Comment by giladgray** [comment]
in fact i might prefer that you reopen this from inside the repo instead of forking so testing locally is simpler...

**Comment by giladgray** [comment]
aaaaand you've been added as a collaborator! mind closing this and re-opening from a non-fork?

File: src/documentation.ts (hunk 3 above)
Comment (ryanmcnamara) [inline src/documentation.ts]: Deferring for this PR

**Comment by ryanmcnamara** [comment]
For the test: it seems like the ts compiler is putting private vars before public

**Pull request merged**